src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -0,0 +1,325 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactTestRendererFiber
+ * @preventMunge
+ * @flow
+ */
+
+'use strict';
+
+var ReactFiberReconciler = require('ReactFiberReconciler');
+var ReactGenericBatching = require('ReactGenericBatching');
+var emptyObject = require('emptyObject');
+
+import type { TestRendererOptions } from 'ReactTestMount';
+
+type ReactTestRendererJSON = {
+  type: string,
+  props: { [propName: string]: string },
+  children: null | Array<string | number | ReactTestRendererJSON>,
+  $$typeof?: any
+}
+
+let instanceCounter = 0;
+
+class TestContainer {
+  createNodeMock: Function;
+
+  constructor(createNodeMock) {
+    this.createNodeMock = createNodeMock;
+  }
+  appendChild(child) {}
+  insertBefore(beforeChild, child) {}
+  removeChild(child) {}
+  toJSON() {}
+}
+
+class TestComponent {
+  id: number;
+  props: Object;
+  type: string;
+  rootContainerInstance: TestContainer;
+  children: Array<Instance | TextInstance>;
+  $$typeof: Symbol;
+
+  constructor(type, props, rootContainerInstance) {
+    this.id = instanceCounter++;
+    this.type = type;
+    this.props = props;
+    this.children = [];
+    this.rootContainerInstance = rootContainerInstance;
+
+    Object.defineProperty(this, 'id', { value: this.id, enumerable: false });
+    Object.defineProperty(this, '$$typeof', {
+      value: Symbol.for('react.test.json'),
+    });
+  }
+
+  appendChild(child) {
+    const index = this.children.indexOf(child);
+    if (index !== -1) {
+      this.children.splice(index, 1);
+    }
+    this.children.push(child);
+  }
+
+  removeChild(child) {
+    const index = this.children.indexOf(child);
+    if (index === -1) {
+      throw new Error('This child does not exist.');
+    }
+    this.children.splice(index, 1);
+  }
+
+  insertBefore(beforeChild, child) {
+    const index = this.children.indexOf(child);
+    if (index !== -1) {
+      this.children.splice(index, 1);
+    }
+    const beforeIndex = this.children.indexOf(beforeChild);
+    if (beforeIndex === -1) {
+      throw new Error('This child does not exist.');
+    }
+    this.children.splice(beforeIndex, 0, child);
+  }
+
+  update(type, props) {
+    this.type = type;
+    this.props = props;
+  }
+
+  toJSON() {
+    const {children, ...props} = this.props;
+    const json: ReactTestRendererJSON = {
+      type: this.type,
+      props: props,
+      children: null,
+    };
+
+    Object.defineProperty(json, '$$typeof', {value: Symbol.for('react.test.json')});
+    if (typeof children === 'string') {
+      json.children = [children];
+    } else {
+      var childrenJSON = [];
+      this.children.forEach((child) => {
+        if (typeof child.toJSON === 'function') {
```

Comment:
Yea, just saying we should clean this up a little so the next person looking isn't more confused than we were.